sdk/resourcemanager/docs/THREAD_POOL_AND_CONNECTION_POOL.md

```diff
@@ -0,0 +1,90 @@
+            .evictInBackground(Duration.ofSeconds(120)) // Every two minutes, the connection pool is regularly checked for connections that are applicable for removal.
+            .build());
+```
+
+### <a name="httpthreadpool"></a>Configure thread pool for the singleton HttpClient:
+By default the TCP client uses an “Event Loop Group”, where the number of the worker threads equals the number of processors available to the runtime on initialization (but with a minimum value of 4). When you need a different configuration, you can use one of the LoopResource#create methods.
+```java readme-sample-azureClientThreadPool
+// Thread pool configuration.
+// Official Reactor Netty documentation for defaults: https://projectreactor.io/docs/netty/release/reference/#client-tcp-level-configurations-event-loop-group
+singletonHttpClientBuilder
+    .eventLoopGroup(LoopResources
+        .create(
+            "client-thread-pool", // thread pool name
+            Runtime.getRuntime().availableProcessors() * 2, // thread pool size
+            true)
+        // we use our custom event loop here, disable the native one
+        .onClient(false))
+    .build();
+```
+
+### <a name="azureidentity"></a>Configure Azure Identity client to use the singleton HttpClient (DefaultAzureCredential for example):
+By default, Azure Identity uses `ForkJoinPool.commonPool()` for token acquisition. The pool size equals the number of processors available to the runtime on initialization minus 1 (with a minimum of 1). It's a singleton pool and is shared by default. Default pool configuration should be sufficient for most use cases.
+```java readme-sample-azureIdentityThreadpool
+// Use the singleton httpClient for Azure Identity
+final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+final TokenCredential credential = new DefaultAzureCredentialBuilder()
+    .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
+    .executorService(ForkJoinPool.commonPool()) // thread pool for executing token acquisition, usually we leave it default
+    .httpClient(singletonHttpClient)
+    .build();
+```
+
+### <a name="azureclient"></a>Configure Azure Clients to use the singleton HttpClient and Azure Identity client:
+```java readme-sample-azureClientHttpClient
+// Use the singleton httpClient for your Azure client
+AzureResourceManager azureResourceManager = AzureResourceManager
+    .configure()
+    .withLogLevel(HttpLogDetailLevel.BASIC)
+    .withHttpClient(singletonHttpClient)
+    .authenticate(credential, profile)
+    .withSubscription(subscriptionId); // your subscription ID, can be different for different Azure clients
+```
+
+### <a name="reactor"></a>Configure Reactor thread pool size:
+By default, Azure Client subscribes on Schedulers.parallel(), which has a thread pool of size equal to available processor count. It is a static pool and is shared in nature.
```

Comment:
Ref to Reactor on `Schedulers.parallel()`?

Reply:
Added for now, still need to check where we introduced the operator.